packages/core/content-manager/server/src/services/document-metadata.ts

```diff
@@ -268,7 +268,13 @@ export default ({ strapi }: { strapi: Core.Strapi }) => ({
     opts: GetMetadataOptions = {}
   ) {
     if (!document) {
-      return document;
+      return {
+        data: document,
+        meta: {
+          availableLocales: [],
+          availableStatus: [],
+        },
+      };
```

Comment:
Saw that if a middleware incorrectly returns an empty result we throw a non descriptive error in here.